scripts/process_new_format.py

```
@@ -171,65 +180,223 @@ def process_markdown_faq_data(markdown_faq_data: dict[str, str]) -> None:
+def extract_faq_source_name(text_content: str) -> Optional[str]:
+    match = FAQ_VERSION_PATTERN.search(text_content)
+    if match:
+        return f"{match.group(1)} v.{match.group(2)}, {match.group(3)}"
+    return None
+
+def extract_related_card_codes(html_content: str, current_card_code: str) -> List[str]:
+    found_codes = set(CARD_LINK_PATTERN.findall(html_content))
+    # Remove the current card's code if it's found, as it's the source_card_code
+    return sorted(list(code for code in found_codes if code != current_card_code))
+
+
```

Ruff (0.11.9)

Line too long (120 > 100)

(E501)